ethpm_types/contract_type.py

```diff
@@ -150,45 +171,57 @@ class SourceMap(BaseModel):
 
     __root__: str
 
+    def __repr__(self) -> str:
+        return self.__root__
+
+    def __str__(self) -> str:
+        return self.__root__
+
     def parse(self) -> Iterator[SourceMapItem]:
         """
-        Parses the source map string into a stream of ``SourceMapItem`` items.
+        Parses the source map string into a stream of
+        :class:`~ethpm_types.contract_type.SourceMapItem` items.
         Useful for when parsing the map according to compiler-specific
         decompilation rules back to the source code language files.
+
+        Returns:
+            Iterator[:class:`~ethpm_types.contract_type.SourceMapItem`]
         """
 
         item = None
 
-        def extract_sourcemap_item(expanded_row, item_idx, previous_val=None):
-            if len(expanded_row) > item_idx and expanded_row[item_idx] != "":
-                return expanded_row[item_idx]
+        def extract_sourcemap_item(_expanded_row, item_idx, previous_val=None):
+            if len(_expanded_row) > item_idx and _expanded_row[item_idx] != "":
+                return _expanded_row[item_idx]
 
-            else:
-                return previous_val  # Use previous item (or None if no previous item)
+            # Use previous item (or None if no previous item).
+            # This is because sourcemaps are compressed to save space,
+            # and this is one of the compression-rules.
+            return previous_val
 
         for i, row in enumerate(self.__root__.strip().split(";")):
 
             if row != "":
                 expanded_row = row.split(":")
 
                 if item is None:
-                    start = int(extract_sourcemap_item(expanded_row, 0) or "-1")
-                    stop = int(extract_sourcemap_item(expanded_row, 1) or "-1")
-                    contract_id = int(extract_sourcemap_item(expanded_row, 2) or "-1")
+                    start = int(extract_sourcemap_item(expanded_row, 0) or -1)
+                    length = int(extract_sourcemap_item(expanded_row, 1) or -1)
+                    contract_id = int(extract_sourcemap_item(expanded_row, 2) or -1)
```

Comment:
Now that the right hand of the `or` is an int should these just read `int(extract_sourcemap_item(expanded_row, 0)) or -1`?

Reply:
We want a falsey string value to be replaced with that number, I don't know what a falsey int value is but `int("")` doesn't work

Reply:
int("" or -1) == -1
int("-1") == -1

I only made this change because it is performance improvement as what as saves a couple unneeded characters (the quotes). Fundamentally, nothing changed here.

Reply:
Oh duh sorry, I misread this! Yeah we want to not run `int("")`